pkg/api/api.go

```diff
@@ -12,6 +12,14 @@ type PaginationData struct {
 	Offset int `query:"offset" json:"offset"` //Offset into the total results
 }
 
+type FilterData struct {
+	Search              string `query:"search" json:"search" `                              //Search string based query to optionally filter-on
+	Arch                string `query:"arch" json:"arch" `                                  //architecture to optionally filter-on
+	Version             string `query:"version" json:"version"`                             //Version to optionally filter-on
+	AvailableForArch    string `query:"available_for_arch" json:"available_for_arch"`       //Filter by compatible arch (IE x86_64 and x86_64 or x86_64 and "Any Arch")
+	AvailableForVersion string `query:"available_for_version" json:"available_for_version"` //Filter by compatible version (IE 7 and 7 or 7 and "Any Version")
```

Comment:
what are these for? how are these different from the other arch/version queries?

Reply:
arch/version filter on those values, the "availableFor" variants basically do the same thing with the caveat that they will also match to "Empty" states, which represent "Any version"/"Any Arch" within the repos.

Reply:
They could, this is a design question though, as we likely want the ability to match to both. It depends on what the API user/UI requires I suppose.

Reply:
Version & Arch are meant to be 'filters' to filter the repo list to a certain state, so you may want to filter 'any arch' vs 'x86_64'.  "AvailableForArch" is meant to be used by other apps like Image Builder that want "All repos applicable for building an image for x86_64", which would include 'x86_64' and 'any arch' repos.  Suggestions welcome for better naming :)

Reply:
I had originally suggested a boolean value for each item to match to blank (any) variants of the filter. This outcome was determined by @jlsherrill. I'm open to a change if it is something that makes sense.

Reply:
I get it now. I don't have a better name for the variable, but maybe the description for it should have an example more like justin's? 'e.g. "x86_64" returns items with "x86_64" or "Any Arch" '

Reply:
It might also make sense to have an example for arch like ' "x86_64" returns repos with "x86_64" `, so there's an obvious distinction between the two.